dotcom-rendering/src/web/components/SubheadingBlockComponent.tsx

```diff
@@ -1,19 +1,36 @@
-import { unwrapHtml } from '../../model/unwrapHtml';
-import { RewrappedComponent } from './RewrappedComponent';
+import { jsx } from '@emotion/react';
+import { JSDOM } from 'jsdom';
+import type { ReactNode } from 'react';
+import { Fragment } from 'react';
 
 type Props = { html: string };
 
+const parseHtml = (html: string): DocumentFragment => JSDOM.fragment(html);
+
+function isElement(node: Node): node is Element {
+	return node.nodeType === 1;
+}
+
+const getAttrs = (node: Node): NamedNodeMap | undefined =>
+	isElement(node) ? node.attributes : undefined;
+
+const buildElementTree = (node: Node): ReactNode => {
+	if (node.nodeType === node.ELEMENT_NODE) {
+		const element = node as Element;
+		return jsx(element.tagName.toLowerCase(), {
+			id: getAttrs(node)?.getNamedItem('id')?.value,
```

Comment:
I think at this point you've already narrowed the type of `node` to `Element`, so you can access `attributes` directly without going via the `getAttrs` function?

With the original code this looks like:

```suggestion
			id: element.attributes.getNamedItem('id')?.value,
```

If using the suggestion above for narrowing `node` this looks like:

```suggestion
			id: node.attributes.getNamedItem('id')?.value,
```